### ISSUE TITLE
DISABLED test_profiler_with_sync_rpc_udf (__main__.RpcTestWithSpawn)

### ISSUE THREAD

**Issue by kshitij12345** [pytorch/pytorch#38008]
https://circleci.com/gh/pytorch/pytorch/5354755?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
```
May 05 14:28:06 ======================================================================
May 05 14:28:06 ERROR [2.071s]: test_profiler_with_sync_rpc_udf (__main__.RpcTestWithSpawn)
May 05 14:28:06 ----------------------------------------------------------------------
May 05 14:28:06 Traceback (most recent call last):
May 05 14:28:06   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 184, in wrapper
May 05 14:28:06     self._join_processes(fn)
May 05 14:28:06   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 286, in _join_processes
May 05 14:28:06     self._check_return_codes(elapsed_time)
May 05 14:28:06   File "/Users/distiller/workspace/miniconda3/lib/python3.7/site-packages/torch/testing/_internal/common_distributed.py", line 319, in _check_return_codes
May 05 14:28:06     raise RuntimeError(error)
May 05 14:28:06 RuntimeError: Processes 1 exited with error code 10
```

cc @ezyang @gchanan @zou3519 @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @xush6528 @jjlilley @osalpekar

**Comment by mrshenli** [comment]
This is likely caused by the same problem as #37557. @rohan-varma and I are working on a fix.

**Comment by rohan-varma** [comment]
Merging https://github.com/pytorch/pytorch/pull/37311 should unblock a fix that's needed to stop these tests from being flaky.

**Comment by mrshenli** [comment]
Fixed by #38352
